zerver/lib/onboarding.py

```diff
@@ -60,6 +71,13 @@ def create_if_missing_realm_internal_bots() -> None:
             setup_realm_internal_bots(realm)
 
 
+def bot_commands(include_help_command: bool = True) -> str:
+    commands = WELCOME_BOT_COMMANDS.copy()
+    if not include_help_command:
+        commands.remove("help")
+    return ", ".join(["`" + command + "`" for command in commands]) + "."
```

Comment:
This pull request makes like 5 independent changes, which makes it very hard to review.  I think this is a good opportunity for you to practice the art of making readable multi-commit PRs; you can split commits using `git rebase -i ` and `git add -p`.

E.g. this change to extract `WELCOME_BOT_COMMANDS` should be a first prep commit, another should be whatever is required as part of adding `transaction.atomic`, etc.  It's soooo much easier to review those individual changes independently (and generally merge them on the first look at a PR), and then have the final commit just be adding the feature and not doing a bunch of other refactoring.

Check out the Zulip commit guidelines for more details: https://zulip.readthedocs.io/en/latest/contributing/version-control.html.

Reply:
Wow, a huge thanks for the example! So I understand that I have to write a story of sorts using the commits, seems interesting, on it!